server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -228,8 +230,23 @@ protected final DocsStats docsStats(IndexReader indexReader) {
             } catch (IOException e) {
                 logger.trace(() -> "failed to get size for [" + info.info.name + "]", e);
             }
+            docsWithIgnoredFields = tryGetNumberOfDocumentsWithIgnoredFields(readerContext);
         }
-        return new DocsStats(numDocs, numDeletedDocs, sizeInBytes);
+        return new DocsStats(numDocs, numDeletedDocs, sizeInBytes, docsWithIgnoredFields);
+    }
+
+    private long tryGetNumberOfDocumentsWithIgnoredFields(final LeafReaderContext readerContext) {
+        try {
+            return readerContext.reader().getSumDocFreq(IgnoredFieldMapper.NAME);
```

Comment:
Should this invoke `getDocCount()` instead? Which returns: `Returns the number of documents that have at least one term for this field`.